client/core-client/src/lib.rs

```diff
@@ -1,57 +1,26 @@
-// Copyright 2019-2021 Parity Technologies (UK) Ltd.
-//
-// Permission is hereby granted, free of charge, to any
-// person obtaining a copy of this software and associated
-// documentation files (the "Software"), to deal in the
-// Software without restriction, including without
-// limitation the rights to use, copy, modify, merge,
-// publish, distribute, sublicense, and/or sell copies of
-// the Software, and to permit persons to whom the Software
-// is furnished to do so, subject to the following
-// conditions:
-//
-// The above copyright notice and this permission notice
-// shall be included in all copies or substantial portions
-// of the Software.
-//
-// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF
-// ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
-// TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
-// PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT
-// SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
-// CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
-// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR
-// IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
-// DEALINGS IN THE SOFTWARE.
-
-use crate::transport::{Receiver as WsReceiver, Sender as WsSender, WsHandshakeError, WsTransportClientBuilder};
-use crate::types::{
-	traits::{Client, SubscriptionClient},
-	v2::{Id, Notification, NotificationSer, ParamsSer, RequestSer, Response, RpcError, SubscriptionResponse},
-	BatchMessage, CertificateStore, Error, FrontToBack, RegisterNotificationMessage, RequestIdManager, RequestMessage,
-	Subscription, SubscriptionKind, SubscriptionMessage, TEN_MB_SIZE_BYTES,
-};
-use crate::{
-	helpers::{
-		build_unsubscribe_message, call_with_timeout, process_batch_response, process_error_response,
-		process_notification, process_single_response, process_subscription_response, stop_subscription,
-	},
-	manager::RequestManager,
+mod helpers;
+mod manager;
+
+use std::time::Duration;
+
+use futures_channel::{mpsc, oneshot};
+use futures_util::{future::Either, SinkExt, StreamExt};
+use jsonrpsee_types::traits::{
+	Client as ClientT, SubscriptionClient as SubscriptionClientT, TransportReceiver, TransportSender,
```

Comment:
Should we just rename the trait to `ClientT`?

Reply:
yeah, sounds good to me.

Reply:
If we do, we should do it to all of the traits I think.